packages/unmock-core/src/generator.ts

```diff
@@ -34,28 +57,16 @@ const setupJSFUnmockProperties = (_: UnmockServiceState) => {
   });
 };
 
-const genMockFromSerializedRequest = (getSpecFromRequest: RequestToSpec) => (
-  sreq: ISerializedRequest,
-): IMock => {
-  const { host } = sreq;
-  // 1. Use sreq to find the proper specification and, as needed, convert
-  //    from short-hand notation to verbose OAS (Mike)
-  // TODO: Link with matcher
-  const spec = getSpecFromRequest(sreq);
-  if (spec === undefined) {
-    throw new Error(`Can't find matching service for '${host}'`);
-  }
-  // 2. Use sreq to fetch the relevant path in the OAS spec
-  // const operation = getPathSchemaFromSpec(spec, method, path);
-  // 3. Fetch state from DSL
+const generateMockFromTemplate = (
+  responseTemplate: any,
```

Comment:
`Response | undefined`?

Reply:
Why would it return undefined if given a template? If something goes wrong, it should throw an error I think.

Reply:
No, the type for `responseTemplate` could be `Response | undefined` instead of `any`.

Reply:
What would empty template mean?

Reply:
Nothing to generate. Empty response.

Reply:
How's that expected in our case? Does OAS allow empty operation object?

Reply:
Technically, only the `responses` top level object is mandatory. All specific responses are optional (`responses: {}` is valid)

Reply:
But `operation: {}` is different than undefined?

Reply:
True, but if we take the route of not throwing errors on failing to match (pt no. 1 in my upper comments), then `undefined` should probably be treated the same (or return `undefined`) and then:
```typescript
    const responseTemplateOrUndefined = serviceStore.match(sreq);

     if (responseTemplateOrUndefined === undefined) {
      return undefined;
    }

     const responseTemplate = responseTemplateOrUndefined;

     const sres = generateMockFromTemplate(responseTemplate);
    return sres;
```
becomes
```typescript
return generateMockFromTemplate(serviceStore.match(sreq));
```

Also, I'm not sure how `js-yaml` specs treat e.g. `responses: null`, that could be another case.

Reply:
Ok, that's just then pushing the undefined check one level deeper imho and could make it harder to debug. But I can change it of course if you think that's prettier.

Reply:
I think it would be prettier and easier if we throw instead of return `undefined` :innocent:

Reply:
Well I don't, this PR is not changing the existing logic so we can make another PR on it later on and let Mike decide.

Reply:
I'd rather we merge so I can continue working and testing from `dev` without conflicts. We can change it, if needed, later on.